src/Smart.Design.Razor/TagHelpers/SideMenu/SideMenuTagHelper.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.TagHelpers;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Smart.Design.Razor.TagHelpers.Constants;
+
+namespace Smart.Design.Razor.TagHelpers.SideMenu;
+
+/// <summary>
+/// Implementation of a Smart design side menu.
+/// Documentation is available <see href="https://design.smart.coop/development/docs/c-side-menu.html">here</see>.
+/// </summary>
+[HtmlTargetElement(TagNames.SideMenuTagName)]
+public class SideMenuTagHelper : TagHelper
+{
+    private readonly ISideMenuHtmlGenerator _sideMenuHtmlGenerator;
+
+    /// <summary>
+    /// Dictionnary of label and links items for the side menu
+    /// </summary>
+    public Dictionary<string, string> LabelAndLinks { get; set; }
```

Comment:
A bit same comment as previous PR, I am not sure it's done on other components (I and Maxime) have written before but I believe we should handle the case where no parameters is passed to the component with an explicit exception
```suggestion
    public Dictionary<string, string> LabelAndLinks { get; set; } = new();
```